### ISSUE TITLE
TypeError: not enough arguments for format string

### ISSUE THREAD

**Issue by yifxiang** [snowflakedb/snowflake-connector-python#103]
Hi, 
I'm using airflow to run some sql scripts. When I try to use query:
select *
from table
where column like '%abc":"def%' 
(% is wildcard for string match)

I got an error: File "/usr/local/lib/python3.6/site-packages/snowflake/connector/cursor.py", line 454, in execute  query = command % processed_params TypeError: not enough arguments for format string

I guess this was caused by some inner config in cursor.py since I could run it on snowflake.

I finally fixed the problem by replacing the first '%' with the exact string. However, that's not a global solution when you do need a wildcard at the beginning of the string.

Could you take a look at this problem and fix it for the future convenience?

Thanks!

**Comment by smtakeda** [comment]
@yifxiang are you binding parameters? `%` is used to bind them if specified for `paramstyle='pyformat'` and `paramstyle='format'`, which are set by default.
https://www.python.org/dev/peps/pep-0249/#paramstyle

If you don't use pyformat, you may want to use `paramstyle='qmark'` in the connect string.

```
con = snowflake.connector.connect(
...
    paramstyle='qmark',
)
```
so that the binding variables are `?` instead of `%`.

**Comment by yifxiang** [comment]
@smtakeda  Thanks for your reply. I did bind parameters with % somewhere else in my script.
i.e: 
select *
from table
where data>%(threshold)s

But the problem is that when I need to use % as a wildcard for string match instead of binding variables (as described in the issue), python will still parse this % as binding variables so that I got sql errors. I wonder is there any methods for me to 'escape' it so that the % only performs as a wildcard?

Moreover, if I change binding variables from % to ?, the same problem will arise when I try to use ? as a symbol for string match instead of binding variables.
